pajbot/apiwrappers/seventv.py

```diff
@@ -2,72 +2,54 @@
 from pajbot.apiwrappers.response_cache import ListSerializer
 from pajbot.models.emote import Emote
 
+from requests import HTTPError
+
+from typing import Tuple
+
 
 class SevenTVAPI(BaseAPI):
     def __init__(self, redis):
-        super().__init__(base_url="https://api.7tv.app/v2/", redis=redis)
+        super().__init__(base_url="https://7tv.io/v3/", redis=redis)
 
     @staticmethod
     def parse_emotes(api_response_data):
-        def get_url(emote_id, size):
-            return f"https://cdn.7tv.app/emote/{emote_id}/{size}x"
+        def get_emote_urls(host) -> Tuple[dict[str, str], int, int]:
+            urls = {}
+            base_url = "https:" + host["url"]
+            emote_size = 1
+            max_width = 0
+            max_height = 0
+            for file in host["files"]:
+                if file["format"] == "WEBP":
+                    name = file["name"]
+                    urls[str(emote_size)] = f"{base_url}/{name}"
+                    if file["width"] > max_width:
+                        max_width = file["width"]
+                    if file["height"] > max_height:
+                        max_height = file["height"]
+                    emote_size += 1
+            return (urls, max_width, max_height)
 
         emotes = []
-        for emote_data in api_response_data:
-            emote_id = emote_data["id"]
+        for active_emote in api_response_data:
```

Comment:
I think I would prefer to keep it at `emote_data`, I've personally never seen somebody refer to the current value in a loop as "active", so naming it that way will just confuse a potential reader

Reply:
7TV refers to emotes in emote-sets as [active emotes](https://github.com/SevenTV/API/blob/e5992c29bb97118bafad1dbe9f4e752aea34d4ea/data/model/emote-set.model.go#L45-L53). The reasoning is that an active emote can have a different name and/or different flags than the base emote (`emote_data`).

Reply:
Ah, I didn't know that. All good then!